internal/domain/user/pg.go

```diff
@@ -0,0 +1,174 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/Azzonya/gophermart/internal/storage"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type Repo struct {
+	Con *pgxpool.Pool
+}
+
+func NewRepo(con *pgxpool.Pool) *Repo {
+	return &Repo{
+		con,
+	}
+}
+
+func (r *Repo) List(ctx context.Context, pars *ListPars) ([]*User, error) {
+	var result []*User
+	var values = make([]interface{}, 0)
+
+	query := "SELECT * FROM users WHERE true"
+
+	if pars.Login != nil {
+		query += " AND login = $1"
+		values = append(values, *pars.Login)
+	}
+
+	if pars.MaxBalance != nil {
+		query += " AND balance <= $2"
+		values = append(values, *pars.MaxBalance)
+	}
+
+	if pars.MinBalance != nil {
+		query += " AND balance <= $3"
+		values = append(values, *pars.MinBalance)
+	}
+
+	rows, err := r.Con.Query(ctx, query, values...)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+	for rows.Next() {
+		var userFound User
+		err = rows.Scan(&userFound.ID, &userFound.Login, &userFound.Password, &userFound.Balance)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, &userFound)
+	}
+
+	return result, err
+}
+
+func (r *Repo) Create(ctx context.Context, obj *GetPars) error {
```

Comment:
```suggestion
func (r *Repo) Create(ctx context.Context, user *User) error {
```
для создания пользователя нам нужен пользователь. Либо если не хочется использовать его с пустыми полями - можно создать отдельный dto 

Аналогично для других запросов, вижу что повсеместно используется одна загадочная структура "GetPars"